### PR TITLE
Code cleaning & Performance improvements

### PR DESCRIPTION
**Pull request by niwinz**
- Remove all unnecessary call forwarding through cats.core namespace.
- Cosmetic fixes such as indentation fixes or namespace aliases renames.
- Use transducers when we can.
- Improve naming.
- Some lightweight core reorganization

### CODE REVIEW

File: src/cats/monad/maybe.cljc [hunk 1]
```diff
@@ -304,14 +302,16 @@
...
 
-;; TODO: use a transducer when we support 1.7.0+
 (defn cat-maybes
-  "Given a collection of maybes, return a sequence of the values that the
-   just's contain."
+  "Given a collection of maybes, return a sequence of the values
+  that the just's contain."
   [coll]
-  (map m/extract (filter just? coll)))
+  (let [xform (comp
```
Comment (dialelo) [inline src/cats/monad/maybe.cljc]: Since it's not a stateful transducer we can even define it outside `cat-maybes`.

**Comment by dialelo** [comment]
Looks great, let me know when it's ready.

**Comment by niwinz** [comment]
@dialelo That do you thinking about the last change (move context into separated namespace).

This will allow write some code that before we can't. Such as import some monads or other times from core without issues about circular dependencies. This is because the majority of imports of cats.core from other namespaces is just for context management.

**Comment by dialelo** [comment]
Makes sense, should we also leave `get-current-context` in core for backwards compatibility? I believe Muse library uses it.

**Comment by niwinz** [comment]
Ok, I'll put an alias for it, thanks for notice me about this.

**Comment by niwinz** [comment]
I think that is already **ready**. So feel free to review and merge! At this moment seems that all tests passes (and no line of test code is changed).

**Pull request merged**